src/workerd/util/sqlite.h

```diff
@@ -26,11 +26,33 @@ using kj::uint;
 // Used to collect periodic metrics about queries and size of sqlite db
 class SqliteObserver {
  public:
+  void setDbWalSize(uint64_t dbWalSize) {
+    this->dbWalSize = dbWalSize;
+  }
+  uint64_t getDbWalSize() {
+    return dbWalSize;
+  }
+  kj::TimePoint now() {
+    return monotonicClock.now();
+  }
   virtual void addQueryStats(uint64_t rowsRead, uint64_t rowsWritten) {}
   // The method is not used by the SqliteDatabase, it is added here for convenience
   virtual void setSqliteStoredBytes(uint64_t sqliteStoredBytes) {}
 
+  virtual void reportQueryEvent(kj::Maybe<kj::String> queryStatement,
+      uint64_t queryRowsRead,
+      uint64_t queryRowsWritten,
+      kj::Duration queryLatency,
+      uint64_t dbWalBytesWritten,
+      int queryError,
```

Comment:
It looks like this parameter value is referred to as `queryResult` in other code?